asdf/extension/_extension.py

```diff
@@ -313,7 +313,13 @@ def __repr__(self):
         else:
             package_description = "{}=={}".format(self.package_name, self.package_version)
 
-        return "<ExtensionProxy class: {} package: {} legacy: {}>".format(
+        if self.extension_uri is None:
+            uri_description = "(none)"
+        else:
+            uri_description = self.extension_uri
+
+        return "<ExtensionProxy URI: {} class: {} package: {} legacy: {}>".format(
+            uri_description,
             self.class_name,
             package_description,
             self.legacy,
```

Comment:
Hey, since we're now python 3.6+, we can use f-strings here.

Reply:
f-f-f-f-fantastic!